messages.go

```diff
@@ -123,3 +123,14 @@ type Data struct {
 	Metadata   map[string]string `json:"metadata"`
 	Content    json.RawMessage   `json:"content"`
 }
+
+// DataMessage is similar to Data, but adds a channel where the response will
+// be send.
+type DataMessage struct {
```

Comment:
This envelope is a side effect of the dispatcher type implementation. I don't think it needs to exist here with the messages that get serialized and transmitted across the network transport. This definition could probably move to grpc.go rather than part of the public yggdrasil package API.